### PR TITLE
SQLiteCipher NSData key support

### PR DESCRIPTION
**Pull request by agisboye**
Adds key and rekey methods that accept a key of type NSData.
This allows the database to be easily keyed with a key generated by
Security.framework:

```swift
let keyData = NSMutableData(length: 64)!
SecRandomCopyBytes(kSecRandomDefault, 64,
UnsafeMutablePointer<UInt8>(keyData.mutableBytes))
db.key(Blob(bytes: keyData.bytes, length: keyData.length))
```

### CODE REVIEW

File: Tests/CipherTests.swift [hunk 1]
```diff
@@ -4,12 +4,22 @@ import SQLiteCipher
...
+        // db
         try! db.key("hello")
 
         try! db.run("CREATE TABLE foo (bar TEXT)")
         try! db.run("INSERT INTO foo (bar) VALUES ('world')")
+        
+        // db2
+        let keyData = NSMutableData(length: 64)!
+        let _ = SecRandomCopyBytes(kSecRandomDefault, 64, UnsafeMutablePointer<UInt8>(keyData.mutableBytes))
+        try! db2.key("hello")
```
Comment (stephencelis) [inline Tests/CipherTests.swift]: Did you mean to key against `keyData`?

**Comment by stephencelis** [comment]
This is great! Found a small typo, but is good to merge otherwise. @mikemee and I are in the process of extracting SQLiteCipher.swift into its own repo, so we'd like to get this in before #314 gets merged!

**Comment by agisboye** [comment]
I knew I'd missed something. Here we go. :-)

**Comment by stephencelis** [comment]
Thanks!

**Pull request merged**